### PR TITLE
Add general node conversion mechanism

### PR DESCRIPTION
**Pull request by hnicke**
Added a general mechanism for node conversion.
Achieved it without a huge switch-case statement, hurray.

Had to add a new csar (empty-topology) in order to test non-parsing relevant features via integration tests, because currently all csars with node templates get rejected (UnsupportedOperationException)

### CODE REVIEW

File: server/src/integration/java/org/opentosca/toscana/core/api/upload/CsarUploadIT.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public void setUp() throws Exception {
     public void testFileUpload() throws Exception {
         System.err.println("Server started!");
 
-        toscanaAPI.uploadCsar("test", TestCsars.CSAR_YAML_VALID_MINIMAL_DOCKER);
+        toscanaAPI.uploadCsar("test", TestCsars.EMPTY_TOPOLOGY);
```
Comment (c-mueller) [inline server/src/integration/java/org/opentosca/toscana/core/api/upload/CsarUploadIT.java]: I assume this is a valid csar?

File: server/src/main/java/org/opentosca/toscana/core/parse/converter/ModelConverter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        Set<RootNode> nodes = convertNodeTemplates(serviceTemplate.getTopologyTemplate());
+        return new EffectiveModel(nodes);
+    }
+
+    private Set<RootNode> convertNodeTemplates(TTopologyTemplateDefinition topology) throws UnknownNodeTypeException {
+        Map<String, TNodeTemplate> templateMap;
+        if (topology != null) {
+            templateMap = topology.getNodeTemplates();
+        } else {
+            logger.warn("Topology template of service template is empty");
```
Comment (c-mueller) [inline server/src/main/java/org/opentosca/toscana/core/parse/converter/ModelConverter.java]: I dont like the sound of this error message. There probably is no real better wording but reading it the first time sounds very confusing

File: server/src/main/java/org/opentosca/toscana/core/parse/converter/NodeConverter.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+import org.eclipse.winery.model.tosca.yaml.TNodeTemplate;
+
+public class NodeConverter {
+
+    public static final String TOSCA_PREFIX = "tosca.nodes.";
+
+    private Map<String, BiFunction<String, TNodeTemplate, RootNode>> conversionMap = new HashMap<>();
+
+    NodeConverter() {
+        addRule("Compute", this::toCompute);
```
Comment (c-mueller) [inline server/src/main/java/org/opentosca/toscana/core/parse/converter/NodeConverter.java]: A short comment Explaining the meaning of these parameters would be good

File: server/src/integration/java/org/opentosca/toscana/core/parse/converter/NodeConverterIT.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            logger.info("Testing conversion of type '{}'", nodeType);
+            TNodeTemplate template = new TNodeTemplate();
+            template.setType(new QName(nodeType));
+            try {
+                converter.convert("name", template);
+                logger.info("Node Type '{}': known", nodeType);
+            } catch (UnsupportedOperationException e) {
+                logger.info("Node Type '{}': known (not yet supported)", nodeType);
+            } finally {
+                System.out.println();
```
Comment (c-mueller) [inline server/src/integration/java/org/opentosca/toscana/core/parse/converter/NodeConverterIT.java]: is this needed?

File: server/src/main/java/org/opentosca/toscana/core/parse/CsarParseServiceImpl.java [hunk 5]
```diff
@@ -6,18 +6,20 @@
...
 import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class CsarParseServiceImpl implements CsarParseService {
 
-    private Logger logger;
+    private Logger logger = LoggerFactory.getLogger(CsarParseService.class);
```
Comment (c-mueller) [inline server/src/main/java/org/opentosca/toscana/core/parse/CsarParseServiceImpl.java]: I think we had the intention of sending the logs to the client if the parsing failed!

File: server/src/integration/java/org/opentosca/toscana/core/api/upload/CsarUploadIT.java (hunk 1 above)
Comment (hnicke) [reply server/src/integration/java/org/opentosca/toscana/core/api/upload/CsarUploadIT.java]: Yeah, will rename it.

File: server/src/integration/java/org/opentosca/toscana/core/parse/converter/NodeConverterIT.java (hunk 4 above)
Comment (hnicke) [reply server/src/integration/java/org/opentosca/toscana/core/parse/converter/NodeConverterIT.java]: Yes, it increases the test output readability

File: server/src/main/java/org/opentosca/toscana/core/parse/CsarParseServiceImpl.java (hunk 5 above)
Comment (hnicke) [reply server/src/main/java/org/opentosca/toscana/core/parse/CsarParseServiceImpl.java]: We will. I will add documentation that explains why this happens.
- as you see in the method beneath, the logger is assigned to the csar's logger.
- this LoggerFactory.getLogger() assignment is actually only needed for testing purposes:
It's possible to call the (also public) method `parse(File)` (which is used internally in `parse(Csar)` for testing purposes. In this case, `logger` would be null.

File: server/src/main/java/org/opentosca/toscana/core/parse/converter/ModelConverter.java (hunk 2 above)
Comment (hnicke) [reply server/src/main/java/org/opentosca/toscana/core/parse/converter/ModelConverter.java]: I will try to reword it to

File: server/src/main/java/org/opentosca/toscana/core/parse/converter/NodeConverter.java (hunk 3 above)
Comment (hnicke) [reply server/src/main/java/org/opentosca/toscana/core/parse/converter/NodeConverter.java]: Fixed
